docs/source/concepts/geometries.rst

```diff
@@ -0,0 +1,109 @@
+Geometries
+==========
+
+To work with our geolocated data in Pyresample we need a way to define that
+geolocation through a combination of the concepts discussed in
+:doc:`geolocated_data` and :doc:`projections`. Pyresample defines geolocation
+through a set of "geometry" objects that fall into two main categories: areas
+and swaths. Areas (a.k.a area definitions) typically represent a grid of
+equally-spaced pixels. A swath of pixels on the other hand
+represents non-uniformly spaced pixels. See the sections below for more details
+on these concepts and how Pyresample understands the information.
+
+Swaths
+------
+
+Swaths are a collection of pixels that may or may not be uniformly sized and
+spaced. The pixels may be contiguous or non-contiguous. In this way swaths of
+pixels can be used to represent any data as a pair of coordinate arrays.
+Typically these coordinates are longitude and latitude arrays in degrees. One
+real world and common case for using swaths to represent data is for low Earth
+orbiting meteorological satellite instruments where due to the scanning pattern
+of the instrument the easiest way to specify the locations of the observed values
+is individual coordinates. In Pyresample, we represent swaths with the
+:class:`~pyresample.geometry.SwathDefinition` class.
+
+For data to be considered contiguous it means that pixels at one location
+in the array are geographically close to the pixels next to them in the array.
+Non-contiguous data is therefore any array where there is no guarantee of
+the geographic location of one pixel relative to any other pixel in the array.
```

Comment:
Also the footprint sizes and shape may vary, and this is not represented in the array in any way.

Reply:
Maybe I just need to reword these top 2 paragraphs for the swath and explicitly talk about contiguity, overlapping, footprint size, and footprint shape.

Reply:
Ok I rewrote this section. I think it is much better and explicit about what someone could expect from a swath.